tests/testsuite/cargo_command.rs

```diff
@@ -365,14 +383,16 @@ fn cargo_subcommand_env() {
     p.cargo("build").run();
     assert!(p.bin("cargo-envtest").is_file());
 
-    let cargo = cargo_exe().canonicalize().unwrap();
     let mut path = path();
     path.push(target_dir.clone());
     let path = env::join_paths(path.iter()).unwrap();
 
     cargo_process("envtest")
         .env("PATH", &path)
-        .with_stdout(cargo.to_str().unwrap())
+        .with_stdout_data(str![[r#"
+[CARGO_PATH]
+
+"#]])
```

Comment:
So this was already a "self updating" test output and I believe the design of it means that there is little to evolve that we'd need to allow snapshotting more from this.

Should we keep the expected-str generation that was in here before, just switching the assert method used?

Reply:
It is not possible to just convert `with_stdout` to `with_stdout_data`, such as the error reported below:
```
--- Expected
++++ actual:   stdout
   1      - /home/heisen/temp/temp/temp/cargo/target/debug/cargo∅
        1 + /home/heisen/temp/temp/temp/cargo/target/debug/cargo
```

Reply:
Adding in a newline is trivial and doesn't invalidate the idea I mentioned that this is unlikely to have surrounding text that gets regularly changed.

Reply:
It has been modified and you can see if it meets the requirements.